src/app/SafeAttributePersistenceProvider.cpp

```diff
@@ -0,0 +1,81 @@
+/*
```

Comment:
This looks like a bad merge of https://github.com/project-chip/connectedhomeip/pull/36473 or something?

Reply:
This is a separate update, #36473 did not have a split dependency between the "safe" persistence and the attribute persistence. The safe one is not ember-specific so it can live in `app` where as the "non-safe" one is ember specific.

I flipped the dependencies so that I can have proper ordering in DataModel::Provider initialization code:
  - I want src/app to not depend on ember, so it cannot initialize a "attributepersistence"
  - attributepersistence requires a "safe attribute persistence" (which can be initialized by Server)
  - Moved the codegen provider to initialize the ember side of the persistence provider.